main.cpp

```diff
@@ -1,95 +1,570 @@
     
-    struct Car                             //4) nested class
+    struct Car //5)                                 Note that the nested type 'Car' is related to the 'CarWash' 
     {
+        //2) member variables with relevant data types.  the names are appropriate for the U.D.T.'s purpose.
         bool isAPickupTruck = false;
-        float gasMileage = 26.2f;        //2) relevant data types
+        float gasMileage = 26.2f;        
+        int year = 1985;
+        std::string manufacturer = "Toyota";
+        std::string model = "Corolla";
+
+        //3) a member function whose parameter has a default value.
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
     };
+
+    //wash and wax car
+    void washAndWaxCar( Car car ); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior( Car car );
     
-    void washAndWaxCar( Car car );         //3) member function with a user-defined type as the parameter.  The user-defined type parameter happens to be the nested class.
-    
-    Car myCar;  //5) a member variable whose type is a UDT.
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
 };
 
 
+
+
 /*
- 1)
+1）Leetcode
+5 properties:
+    1)number of comments
+    2)number of submissions
+    3)number of solved questions
+    4)acceptance rate
+    5)overall score
+3 things it can do
+    1)write code
+    2)help others with their code
+    3)pin a thread
  */
-
+struct Leetcode
+{
+    //1)number of comments
+    int numberOfComments = 24;
+    //2)number of submissions
+    int numberOfSubmissions = 4;
+    //3)number of solved questions
+    int numberOfSolvedQuestions = 13;
+    //4)acceptance rate
+    float acceptanceRate = 0.66f;
+    //5)overall score
+    float overallScore = 87.5f;
+    //1)write code
+    void writeCode(int languageType);
+    //2)help others with their code
+    void helpOthersWithTheirCode(int numberOfQuestion);
+    //3)pin a thread
+    void pinAThread(int numberOfThread);
+};
 /*
- 2)
+2)Safeway
+5 properties:
+    1)open time
+    2)discount
+    3)Customer's points
+    4)close time
+    5)number of parking spots
+3 things it can do
+    1)order grocery online
+    2)refund
+    3)collect coupons
  */
-
+struct Safeway
+{
+    //1)open time
+    float openTime = 9.f;
+    //2)discount
+    float discount = 0.7f;
+    //3)Customer's points
+    int customerPoints = 234;
+    //4)close time
+    float closeTime = 21.f;
+    //5)number of parking spots
+    int numberOfParkingSpots = 34;
+    struct Customer
+    {
+        int accountnumber = 133423;
+        int membership = 2;
+        bool relatedCreditCard = 0;
+    };
```

Comment:
Nested structs need to have 5 properties (member variables) and 3 actions (functions)
- [ ] revise